scipy/special/_basic.py

```diff
@@ -2372,51 +2447,54 @@ def factorial(n, exact=False):
     120
 
     """
-    if exact:
-        if np.ndim(n) == 0:
-            if np.isnan(n):
-                return n
-            return 0 if n < 0 else math.factorial(n)
-        else:
-            n = asarray(n)
-            un = np.unique(n).astype(object)
-
-            # Convert to object array of long ints if np.int_ can't handle size
-            if np.isnan(n).any():
-                dt = float
-            elif un[-1] > 20:
-                dt = object
-            elif un[-1] > 12:
-                dt = np.int64
+    # handle array-likes
+    n = n if np.isscalar(n) else asarray(n)
+    if isinstance(n, np.ndarray):
+        if not (np.issubdtype(n.dtype, np.integer)
```

Comment:
Use `np.inexact` for these tests. complex nums are `np.cfloating` as an example.

Reply:
The [documentation](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.inexact) for `np.inexact` doesn't mention what exactly is included there, and neither does the link to the sources.

I want to restrict this specifically to integers and (real) floats - how would I do that using `np.inexact`?

Reply:
This question is still open, if you have some cycles...